### PR TITLE
Update incremental-models.md

### LINKED ISSUE #3538: BUG - Incremental models doc incorrect since changes in #3530
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/docs/build/incremental-models#supported-incremental-strategies-by-adapter

### What part(s) of the page would you like to see updated?

There were changes in the PR #3530

The entry for `dbt-snowflake` is not correct anymore ; it should be `merge` and `delete+insert` not `append`. I haven't checked in details all the other adapters.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mirnawong1**
bug issue reported by @b-per with regards to recent pr #3530. cc @dataders @dbeatty10 

Resolves #3538

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `website/static/_redirects`
- [ ] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

File: website/docs/docs/build/incremental-models.md [hunk 1]
```diff
@@ -260,9 +260,9 @@ Click the name of the adapter in the below table for more information about supp
 | dbt-postgres                                                                                     | `append` | `delete+insert`                                         |
 | dbt-redshift                                                                                     | `append` | `delete+insert`                                         |
 | [dbt-bigquery](/reference/resource-configs/bigquery-configs#merge-behavior-incremental-models)   | `merge`  | `insert_overwrite`                                      |
-| [dbt-spark](/reference/resource-configs/spark-configs#incremental-models)                        | `append` | `merge` (Delta only)                                    |
+| [dbt-spark](/reference/resource-configs/spark-configs#incremental-models)                        | `append` | `merge` (Delta only), `insert_overwrite`, `partition_by`|
```
Comment (mirnawong1) [inline website/docs/docs/build/incremental-models.md]:
```suggestion
| [dbt-spark](/reference/resource-configs/spark-configs#incremental-models)                        | `append` | `merge` (Delta only), `insert_overwrite` |
```

File: website/docs/docs/build/incremental-models.md [hunk 2]
```diff
@@ -260,9 +260,9 @@ Click the name of the adapter in the below table for more information about supp
 | dbt-postgres                                                                                     | `append` | `delete+insert`                                         |
 | dbt-redshift                                                                                     | `append` | `delete+insert`                                         |
 | [dbt-bigquery](/reference/resource-configs/bigquery-configs#merge-behavior-incremental-models)   | `merge`  | `insert_overwrite`                                      |
-| [dbt-spark](/reference/resource-configs/spark-configs#incremental-models)                        | `append` | `merge` (Delta only)                                    |
+| [dbt-spark](/reference/resource-configs/spark-configs#incremental-models)                        | `append` | `merge` (Delta only), `insert_overwrite` |
 | [dbt-databricks](/reference/resource-configs/databricks-configs#incremental-models)              | `append` | `merge` (Delta only) `insert_overwrite` `delete+insert` |
```
Comment (mirnawong1) [inline website/docs/docs/build/incremental-models.md]:
```suggestion
| [dbt-databricks](/reference/resource-configs/databricks-configs#incremental-models)              | `append` | `merge` (Delta only) `insert_overwrite` |
```

File: website/docs/docs/build/incremental-models.md [hunk 3]
```diff
@@ -260,9 +260,9 @@ Click the name of the adapter in the below table for more information about supp
...
 | [dbt-bigquery](/reference/resource-configs/bigquery-configs#merge-behavior-incremental-models)   | `merge`  | `insert_overwrite`                                      |
-| [dbt-spark](/reference/resource-configs/spark-configs#incremental-models)                        | `append` | `merge` (Delta only)                                    |
-| [dbt-databricks](/reference/resource-configs/databricks-configs#incremental-models)              | `append` | `merge` (Delta only) `insert_overwrite` `delete+insert` |
-| [dbt-snowflake](/reference/resource-configs/snowflake-configs#merge-behavior-incremental-models) | `merge`  | `append`                                                |
+| [dbt-spark](/reference/resource-configs/spark-configs#incremental-models)                        | `append` | `merge` (Delta only), `insert_overwrite` |
+| [dbt-databricks](/reference/resource-configs/databricks-configs#incremental-models)              | `append` | `merge` (Delta only) `insert_overwrite` |
+| [dbt-snowflake](/reference/resource-configs/snowflake-configs#merge-behavior-incremental-models) | `merge`  | `delete+insert`                                         |
```
Comment (dataders) [inline website/docs/docs/build/incremental-models.md]:
```suggestion
| [dbt-spark](/reference/resource-configs/spark-configs#incremental-models)                        | `append` | `merge` (Delta only)  `insert_overwrite` |
| [dbt-databricks](/reference/resource-configs/databricks-configs#incremental-models)              | `append` | `merge` (Delta only) `insert_overwrite` |
| [dbt-snowflake](/reference/resource-configs/snowflake-configs#merge-behavior-incremental-models) | `merge`  | `append`, `delete+insert`                                        |
```

**Review by dataders — APPROVED** [review]
thanks @mirnawong1 and @b-per!

**Pull request merged**